build-system/tasks/helpers.js

```diff
@@ -571,12 +586,26 @@ async function minify(code, map) {
       beautify: !!argv.pretty_print,
       // eslint-disable-next-line google-camelcase/google-camelcase
       keep_quoted_props: true,
+      // // TODO: only add preamble for mainBundles?
       preamble: ';',
     },
     sourceMap: {content: map},
     module: !!argv.esm,
   };
 
+  // Enabling property mangling requires disabling two other optimization.
+  // - Should not mangle computed properties (often used for cross-binary purposes)
+  // - Should not convert computed properties into regular property access (compress.properties = false)
+  if (mangle) {
+    // eslint-disable-next-line google-camelcase/google-camelcase
+    terserOptions.mangle.properties = {keep_quoted: true, regex: '_$'};
+    terserOptions.nameCache = nameCache;
+
+    // Disables converting computed properties (a['hello']) into regular prop access (a.hello).
+    // This was an assumption baked into closure.
+    terserOptions.compress.properties = false;
```

Comment:
Why do we have to disable this if it's the same system doing mangling and inlining?

Reply:
Great question! I've forgotten the specific example that fails (will take me awhile to reproduce it again), but I can demonstrate the idea.

Imagine this snippet:
```js
const RESOURCE_PROP = '__AMP__RESOURCE_';
if (x[RESOURCE_PROP]) {
  const r = x[RESOURCE_PROP];
  doStuff(r);
}
```

If we perform an inline and compress pass, we are left with:
```js
if (x.__AMP__RESOURCE_) {
  const r = x.__AMP__RESOURCE_;
  doStuff(r);
}
```

Because the string value happened to end with a `_`, our regex will flag it as eligible for mangling. This won't work across binaries and may even have issues within the same binary if some accesses of the property maintain their string form

Reply:
This seems like a bug in terser, really. It's probably going from `x[RESOURCE_PROP]` directly to `x.__AMP_RESOURCE_` instead of first replacing to `x['__AMP_RESOURCE_']` then `x.__AMP_RESOURCE_`.

Reply:
I just compiled the codebase with the flag enabled and disabled, and it didn't appear to make any difference. Are you sure this is needed?

Reply:
Found it. Use `computed_props = false` instead, to perform better minification overall while still fixing this issue.

Reply:
Found the issue. Just disable `computed_props`.
```suggestion
    terserOptions.compress.computed_props = false;
```

Reply:
Will see if this works

Reply:
What was the issue you found?

Reply:
switching to `compued_props` shaved another 0.3kb brotli